lib/rust/parser/src/source/code.rs

```diff
@@ -13,98 +13,133 @@ use crate::prelude::*;
 #[derive(Debug, Clone, Default, Eq, PartialEq, Deref)]
 pub struct StrRef<'s>(pub &'s str);
 
-/// A code representation. It can either be a borrowed source code or a modified owned one.
+/// Identifies a location in source code.
+#[derive(
+    Copy,
+    Clone,
+    Debug,
+    Default,
+    Eq,
+    PartialEq,
+    Serialize,
+    Reflect,
+    Deserialize,
+    PartialOrd,
+    Ord
+)]
+pub struct Location {
+    /// Offset from the beginning, in UTF-8 code units (bytes).
+    #[reflect(hide)]
+    pub utf8:  u32,
+    /// Offset from the beginning, in UTF-16 code units (two-byte words).
+    #[reflect(hide)]
+    pub utf16: u32,
+    /// Offset from the first line.
+    #[reflect(hide)]
+    pub line:  u32,
```

Comment:
The "offset" is not clear enough here. Also it is not really clear what is a "line" really. In LSP, the lines are strictly defined to be separated by a numer of possible EOL characters. Therefore the file line ending style doesn't influence the locations at all. We should probably adapt exactly the same semantics. https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocuments

Reply:
Fixed it to match LSP (which matches the rest of the language syntax).